docs/explainer.md

```diff
@@ -34,4 +34,32 @@ let writableStream = tcpSocket.writable;
 tcpSocket.close();
 ```
 
+To simplify security analysis, an API for listening for incoming connections is not yet being proposed.
+
 ## UDP
+
+To simplify security analysis, the initial proposal only supports cases where the web app initiates communication with a remote host.
+
+```
+const options = {
+    remoteAddress: 'example.com',
+    remotePort: 7
+};
+navigator.openUDPSocket(options).then(udpSocket => { ... }).else(error => { ... });
+```
+
+The UDP socket can use events to indicate when a packet can be sent or received. No send event is generated until after a packet has been sent.
```

Comment:
Any reason events were chosen over a stream?

Reply:
By "stream" do you mean a [Streams API stream](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)? Or some other more generic concept of an iterable sequence?

We don't want the Streams API for UDP, since that represents a continuous stream of bytes, whereas UDP is broken into discrete packets, which are important to capture in the API.

We'd want it to be a lazy iterable object that yields Packet objects.

It looks like JavaScript has a concept of a [generic iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols), and it's exposed through WebIDL as [`async iterable`](https://heycam.github.io/webidl/#idl-async-iterable). If that's what you have in mind, I'm on board with this, though I have never written something using this concept (I'm familiar with how it works in Python though).

Reply:
I had considered using Promises instead of events: each arriving packet could resolve a promise, and each packet send resolve a promise as soon as the system had buffer space available for another packet to be sent.

I wasn't aware of `async iterable`. That could work for receiving. We would still have to choose between events and promises for sending. I think events make most sense: a common use case it to send as soon as new information is available, or send status updates periodically, but allow for the possibility of the send buffer filling, and it such cases wait for buffer space to be available and then send the most current information.

Reply:
> We would still have to choose between events and promises for sending.

Maybe I don't fully understand how UDP works. Why is there an "`onsend`" event? Sending is something you *do*, not something that happens.

I'm guessing "`onsend`" is really "`onreadytosend`" (i.e., the port is unblocked). It seems that it would be more natural to allow `send()` to be called at any time, but returns a promise that resolves once the port is unblocked and the packet is sent. Internally, it's waiting for the port to be unblocked, but the user doesn't need to know about that.